velox/exec/Spill.h

```diff
@@ -0,0 +1,350 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "velox/common/file/File.h"
+#include "velox/exec/TreeOfLosers.h"
+#include "velox/vector/ComplexVector.h"
+#include "velox/vector/DecodedVector.h"
+#include "velox/vector/FlatVector.h"
+
+namespace facebook::velox::exec {
+
+// A lightweight, non-owning reference to a row in a RowVector read from
+// spill.
+struct SpillFileRow {
+  // Decodes each column of 'rowVector'. The decoding is done right after
+  // reading 'rowVector' from a spill file.
+  std::vector<std::unique_ptr<DecodedVector>>* decoded;
+  // A batch of rows read from a spill file.
+  RowVector* rowVector;
+  // Index of the current row in 'rowVector'.
+  vector_size_t index;
+};
+
+// Input stream backed by spill file.
+class SpillInput : public ByteStream {
+ public:
+  // Reads from 'input' using 'buffer' for buffering reads.
```

Comment:
What's the purpose for passing in the buffer? Why can't this class allocate its own buffer? Is this for memory re-use?